### DISCUSSION TITLE
What's the interpolator ?

### DISCUSSION THREAD

**Discussion by fjgaston** [garygru/yummyDSP#3] [Q&A] [answered]
Hi, maybe I have limited knowledge in C++ but I don't understand what is the interpolator? What is its purpose and how does it work?
I don't understand the use of kCurrent, kTarget and KSmoothCount.
I forked your project and already created a new function in the Delay Node, the function is called setDelayFeedback()
[https://github.com/fjgaston/yummyDSP/blob/master/src/Nodes/DelayNode.cpp](https://github.com/fjgaston/yummyDSP/blob/master/src/Nodes/DelayNode.cpp)
In the near future I'd like to add setWet(); and setDry(); (because it makes more sense than setMix() in the project that I have. But I run into these kCurrent and kTarget in the setMix() function that I don't really know how to deal with.

**Comment by garygru** [comment, marked as answer]
Hey, the interpolator smoothes the parameters so that there are no audible clicking noises when changing a value. kTarget is the target value (where you want to go) and kCurrent describes the current value (where you are right now). At some point e.g. after 50ms the current value has ramped to the target value. 
This means you need to set the target value and read the current value. The interpolator does the rest for you. Does this make sense?

**Comment by fjgaston** [comment]
hi, thanks.
Yes it makes sense and the "fade" parameter in the setDelayMs() function also make more sense to me now.
